### PR TITLE
Use query params and make returned fields selectable (GSI-940)

### PR DESCRIPTION
**Pull request by Cito**
- use GET instead of POST for the `search` endpoint
  - this allows caching responses and
  - avoids problems with missing CSRF tokens which the auth adapter does for POST requests when there is a session
  - CSRF is not needed since searching has no side effects
- use query params instead of passing the search parameters in the body
  - since GET requests should not have a body
- renamed "properties" to "fields"
   - since this is shorter and more Mongo document specific
- remove "rpc" prefix for the two provided endpoints
   - since it usually implies POST and
   - any prefix can be added via the api root path anyway
- add a new config option `selected_field`
  - if empty, all fields are returned in the result (backward compatible behavior)
  - if not empty, only the specified fields are returned under `hits` 
  - can also be used to pass the proper labels for these fields to the client
- refactor sorting tests
  - make tests more rigorous and test more cases
  - particularly test sorting with fields that are not selected
- add metadata from the distribution to the OpenAPI doc
  - we should put something like this in the microservice template or service commons
- bump the major version since these are breaking changes

### CODE REVIEW

**Review by TheByronHimes — CHANGES_REQUESTED** [review]
A couple small things included here, but the primary issue I encountered was that including `selected_fields` config caused both `test_relevance` and `test_sorting` to fail. Probably something to do with the added projection occurring before the sort.

File: openapi.yaml [hunk 1]
```diff
@@ -237,8 +175,8 @@ components:
       title: ValidationError
       type: object
 info:
-  title: FastAPI
-  version: 0.1.0
+  title: A service for searching metadata artifacts and filtering results.
```
[the remark this answers is not in the document]
Comment (Cito) [reply openapi.yaml:178]: The abbreviation "mass" does not make a good title. But I agree, the long form (abbreviation written out) would be better a better, right. This is more like the summary.

File: src/mass/adapters/inbound/fastapi_/configure.py [hunk 2]
```diff
@@ -18,13 +18,23 @@
...
 
 
 def get_configured_app(*, config: Config) -> FastAPI:
     """Create and configure a REST API application."""
-    app = FastAPI()
+    doc = mass.__doc__
+    try:
+        title, description = mass.__doc__.split("\n\n", 1)
+    except ValueError:
+        title, description = doc, ""
```
[the remark this answers is not in the document]
Comment (Cito) [reply src/mass/adapters/inbound/fastapi_/configure.py:32]: Actually, OpenAPI provides three fields: title, summary, description. The title could be the long form of the name, the summary could be what currently is in the docstring. And the description is even more detailed, something that could be taken from a README section. Or it could simply contain a link to the repo.

Will change this.

File: openapi.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (TheByronHimes) [reply openapi.yaml:178]: I see, that makes sense

File: .readme_generation/description.md
[the remark this answers is not in the document]
Comment (Cito) [reply .readme_generation/description.md:1]: Can you make a suggestion?

File: openapi.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Cito) [reply openapi.yaml]: This is changed now in ef00a98.

File: src/mass/adapters/inbound/fastapi_/configure.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Cito) [reply src/mass/adapters/inbound/fastapi_/configure.py]: Changed now in ef00a98. Instead of using docstrings, we can use the more public distribution metadata.

Maybe we can put something like this also in the microservice template or service commons.

**Comment by Cito** [comment]
I have refactored the sorting tests now to check all of the edge cases, particularly when sorting with unselected fields.

File: src/mass/adapters/inbound/fastapi_/configure.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (TheByronHimes) [reply src/mass/adapters/inbound/fastapi_/configure.py]: @Cito given the near-identical contents of that function, it would be nice if there was a standardized function for it instead of copy-pasta in every fastapi service.
